lib/charms/observability_libs/v1/kubernetes_service_patch.py

```diff
@@ -158,6 +180,8 @@ def __init__(
             additional_selectors: Selectors to be added to the kubernetes service (by default only
                 "app.kubernetes.io/name" is set to the service name)
             additional_annotations: Annotations to be added to the kubernetes service.
+            refresh_event: an optional bound event or list of bound events which
+                will be observed to re-apply the patch (eg on port change)
```

Comment:
But if user changes config, then install is not fired. I don't see a way to reapply the patch outside of this lib

Reply:
I'm saying that you probably don't even need `install`: seems like all logic can be run at once in the constructor.
But this would be a deviation from current practice and is outside the scope of this PR :)

Reply:
ahh, I see